docs/getting-started/setup/index.md

```diff
@@ -14,45 +14,194 @@ helm repo update
+```
+
+In order for traffic to enter the Istio mesh, the request needs to go through an Istio ingress
+gateway, which is simply a normal Kubernetes Deployment and Service. One convenient way to reach
+the gateway using minikube, is using the `minikube tunnel` command which assigns Services a 
+LoadBalancer. This command should be run in the background, usually in a separate terminal window:
+
+```shell
+minikube tunnel
+```
+
+While running `minikube tunnel`, the `istio-ingressgateway` Service will now have an external IP
+which can be retrieved via `kubectl`:
+
+```shell
+$ kubectl get svc -n istio-system istio-ingressgateway
+NAME                   TYPE           CLUSTER-IP      EXTERNAL-IP     PORT(S)                            AGE
+istio-ingressgateway   LoadBalancer   10.100.136.45   10.100.136.45   15020:31711/TCP,80:31298/TCP....   7d22h
+```
+
+The LoadBalancer external IP (10.100.136.45 in this example) is now reachable to access services in
+the Istio mesh. Istio routes requests to the correct pod based on the Host HTTP header. Follow the
+guide on [supplying host headers](#supplying-host-headers) to learn how to configure your client
+environment to supply the proper request to reach the pod.
+
+## Supplying Host Headers
 
 Most ingress controllers and service mesh implementations rely on the 
 [Host HTTP request header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host) being
-supplied in the request, in order to determine how to route the request to the correct service.
-There are different ways to ensure the host is supplied with the request. In real, production
-environments, this is typically achieved by adding entry for the hostname a DNS server.
+supplied in the request in order to determine how to route the request to the correct pod.
+
+### Determining the hostname to IP mapping
 
-On local workstations, we can sidestep the DNS changes by adding an entry to `/etc/hosts`,
-to map the hostname and IP address of the ingress. First obtain the values from the ingress:
+For the Host header to be set in the request, the hostname of the service should resolve to the
+public IP address of the ingress or service mesh. Depending on if you are using a ingress controller
+or a service mesh, use one of the following techniques to determine the correct hostname to IP
+mapping:
+
+#### Ingresses
+
+For traffic which is reaching the cluster network via a normal Kubernetes Ingress, the hostname
+should map to the IP of the ingress. We can retrieve the external IP of the ingress from the
+Ingress object itself, using `kubectl`:
 
 ```shell
 $ kubectl get ing rollouts-demo-stable
 NAME                   CLASS    HOSTS                 ADDRESS        PORTS   AGE
 rollouts-demo-stable   <none>   rollouts-demo.local   192.168.64.2   80      80m
 ```
 
-From the above output, the correct entry in `/etc/hosts` would be:
+In the example above, the hostname `rollouts-demo.local` should be configured to resolve to the
```

Comment:
We should indicate that we are going to explain how to resolve that hostname correctly below